src/Product/AttributeMapping/AttributeMappingHelper.php

```diff
@@ -0,0 +1,316 @@
+				$sources,
+				[
+					'disabled:taxonomies' => __( '- Taxonomies -', 'google-listings-and-ads' ),
+				],
+				$parsed_taxonomies
+			);
+		}
+
+		return $sources;
+	}
+
+	/**
+	 * Get a list of the available product fields.
+	 *
+	 * @return array An array with the available product fields.
+	 */
+	public static function get_source_product_fields() {
+		$fields = [
+			'product:backorders'       => __( 'Allow backorders setting', 'google-listings-and-ads' ),
+			'product:product_title'    => __( 'Product title', 'google-listings-and-ads' ),
+			'product:sku'              => __( 'SKU', 'google-listings-and-ads' ),
+			'product:stock_qty'        => __( 'Stock Qty', 'google-listings-and-ads' ),
+			'product:stock_status'     => __( 'Stock Status', 'google-listings-and-ads' ),
+			'product:tax_class'        => __( 'Tax class', 'google-listings-and-ads' ),
+			'product:variation_title'  => __( 'Variation title', 'google-listings-and-ads' ),
+			'product:weight'           => __( 'Weight (raw value, no units)', 'google-listings-and-ads' ),
+			'product:weight_with_unit' => __( 'Weight (with units)', 'google-listings-and-ads' ),
+		];
+		asort( $fields );
+
+		$fields = array_merge(
+			[
+				'disabled:product' => __( '- Product fields -', 'google-listings-and-ads' ),
+			],
+			$fields
+		);
+
+		return apply_filters( 'woocommerce_gla_attribute_mapping_fields_product_fields', $fields );
+	}
+
+	/**
+	 * Returns the available meta (non system) metas
+	 *
+	 * @return array The available metas
+	 */
+	public static function get_source_custom_fields() {
+		global $wpdb;
+		$transients = self::get_transients_object();
+
+		$cached_metas = $transients->get(
+			TransientsInterface::ATTRIBUTE_MAPPING_META_FIELDS,
+		);
+
+		if ( $cached_metas ) {
+			return $cached_metas;
+		}
+
+		$sources = [];
+
+		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
+		$sql = $wpdb->prepare(
+			"SELECT DISTINCT {$wpdb->postmeta}.meta_key
+		                FROM {$wpdb->posts}
+			       LEFT JOIN {$wpdb->postmeta}
+			              ON {$wpdb->posts}.ID = {$wpdb->postmeta}.post_id AND {$wpdb->postmeta}.meta_key NOT LIKE %s
+				       WHERE {$wpdb->posts}.post_type IN ( %s, %s )",
+			[ $wpdb->esc_like( '_' ) . '%', 'product_variation', 'product' ]
+		);
+
+		$meta_keys = $wpdb->get_col( $sql );
+		foreach ( $meta_keys as $meta_key ) {
+			$sources[ 'meta:' . $meta_key ] = $meta_key;
+		}
+
+		if ( ! empty( $sources ) ) {
+			$sources = array_merge(
+				[
+					'disabled:meta' => __( '- Custom fields -', 'google-listings-and-ads' ),
+				],
+				$sources
+			);
+		}
+
+		$transients->set(
+			TransientsInterface::ATTRIBUTE_MAPPING_META_FIELDS,
+			$sources,
+			HOUR_IN_SECONDS
+		);
+
+		return apply_filters( 'woocommerce_gla_attribute_mapping_fields_custom_fields', $sources );
+	}
+
+	/**
+	 * Allowing to register custom attributes by using a filter.
+	 *
+	 * @return array The custom attributes
+	 */
+	public static function get_source_custom_attributes() {
```

Comment:
Those Custom attributes are under the umbrella of custom attributes and have the prefix `attribute` that we will use to get the attribute by calling:

`$product->get_attribute( $attribute_key );`

In case of Yoast has their own prefixes because we need to get the Yoast attributes differently.